src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +86,69 @@ private Menu createFileMenu() {
+    public final boolean isCurrentOpenBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a {@code ConfirmChangesDialog} to ask the user whether or not to save the unsaved changes
+     * in the current board.
      */
-    public final void newBoard() {
-        if (!isNewBoardCreationDialogConfirmed()) {
-            logger.info("New board creation cancelled");
+    public final void promptToSaveCurrentBoard() {
+        ConfirmChangesDialog dlg = new ConfirmChangesDialog(mainStage);
+        Optional<ButtonType> response = dlg.showAndWait();
+        if (response.get() != ButtonType.YES) {
+            logger.info("User abandoned unsaved changes.");
             return;
         }
 
-        panels.closeAllPanels();
-        saveBoardAs();
+        saveBoard();
+        logger.info("Changes to the current board saved.");
     }
 
     /**
-     * Returns a boolean indicating whether we can proceed with
-     * the new board creation. If there are no panels open, it
-     * returns true; else a dialog is shown for the user to confirm.
+     * Prompts a {@code BoardNameDialog} to ask the user for a new board name
+     * @return the user's response
      */
-    private boolean isNewBoardCreationDialogConfirmed() {
-        List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-        if (panelList.isEmpty()) {
-            return true;
+    public final Optional<String> promptForNewBoardName() {
+        BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
+        Optional<String> response = dlg.showAndWait();
+        ui.showMainStage();
+        panels.selectFirstPanel();
+        return response;
+    }
+
+    /**
+     * Logs the user action and creates a new board
+     */
+    private void onBoardNew() {
+        logger.info("Menu: Boards > New");
+        if (isCurrentOpenBoardDirty()) {
+            promptToSaveCurrentBoard();
         }
 
-        Alert dlg = new Alert(AlertType.CONFIRMATION, "");
-        dlg.initModality(Modality.APPLICATION_MODAL);
-        dlg.setTitle("Confirmation");
-        dlg.getDialogPane().setHeaderText("Create a new board");
-        dlg.getDialogPane().setContentText("Are you sure you want to create a new board?"
-                                           + " All unsaved changes to the current board will be lost.");
-        Optional<ButtonType> response = dlg.showAndWait();
-        return response.isPresent() &&
-                response.get().getButtonData() == ButtonData.OK_DONE;
+        newBoard();
+    }
+
+    /**
+     * Creates an empty board. User will be prompted to
+     * confirm the action if there are unclosed panels
+     */
+    public final void newBoard() {
```

Comment:
Actually it is not really an "empty" board. It has an empty panel by default. Coz I think it is not useful to create just an empty board, the first thing that the user will do is to add an empty board to it. So why don't we just do that for the user?